rfcs/text/0031-risk-fields.md

```diff
@@ -21,6 +21,18 @@ Stage X: Provide a brief explanation of why the proposal is being marked as aban
 <!--
 Stage 1: Describe at a high level how this change affects fields. Include new or updated yml field definitions for all of the essential fields in this draft. While not exhaustive, the fields documented here should be comprehensive enough to deeply evaluate the technical considerations of this change. The goal here is to validate the technical details for all essential fields and to provide a basis for adding experimental field definitions to the schema. Use GitHub code blocks with yml syntax formatting, and add them to the corresponding RFC folder.
 -->
+The `risk` fields being proposed are as follows:
+
+Field | Type | Example | Description | Use Case
+-- | -- | -- | -- | -- 
+risk.score | float | 88.73 | A score in the range 0-100 that quantifies the amount of risk associated with an entity. The higher the score, the higher the risk. | Can be used to indicate the risk associated with a particular host
+risk.level | keyword | High | A risk level corresponding to the risk score | Can be used to indicate the risk associated with a particular host
+risk.info.*| group | - | Information about what contributed to the risk | Rules that made a host high-risk
```

Comment:
Can you elaborate more here on what is to be expected? Were you thinking something with the `[rule.*](https://www.elastic.co/guide/en/ecs/current/ecs-rule.html)` field set?

Reply:
I was thinking a reusable field set that can be used with any asset/entity. Having a separate reusable field set for `risk` also helps us standardize the definition of risk across assets. So these fields would be used with field sets such as (but not limited to) `rule.*`, `host.*`, `user.*`, `event.*`. This might mean updated to the existing risk-related fields currently present in some field sets, for example `event.risk_score` would become `event.risk.score` etc.

Reply:
Gotcha. For the field `risk.info.*`, what exactly did you have in mind? I see it listed as type `group`, but I don't think that is an actual ES type, and I see it ends in .* leading me to believe it's an object/field set.

Reply:
@kgeller I noticed [another RFC](https://github.com/elastic/ecs/blob/risk-fields-stage-1/rfcs/text/0027-faas-fields.md) that I was using as reference, which had a type `group`. Looks like they've updated that though. But yes, the intent was for the `risk.info.*` fields to be an object. This field should contain explainability about the calculated risk score for a particular asset, but these explainability metrics could look different, for different assets. I wasn't sure how to capture that- any inputs on this would be super helpful!

Reply:
Will that object be something new, or were you planning on re-using an existing field set such as [`rule.*`](elastic.co/guide/en/ecs/current/ecs-rule.html)?

Reply:
This will be new. I'm going to update this a bit.